eng/pipelines/templates/jobs/vmr-build.yml

```diff
@@ -547,13 +554,22 @@ jobs:
 
           if [[ "$prepSdk" == "false" ]]; then
             customPrepArgs="${customPrepArgs} --no-sdk"
-            rm -rf $(sourcesPath)/.dotnet
-            mkdir $(sourcesPath)/.dotnet
-            previousSdkPath="$(sourcesPath)/prereqs/packages/archive/dotnet-sdk-*.tar.gz"
-            eval tar -ozxf "$previousSdkPath" -C "$(sourcesPath)/.dotnet"
-            eval rm -f "$previousSdkPath"
+            
```

Comment:
I think this logic to extract the sdk and package should be moved to the `prep the Build` step.  Both the `prep-source-build.sh` and `build.sh` should be passing in `--with-sdk` and `--with-packages` when the appropriate parameters are set.  If you look at the current prep logic, you see it is trying to use the downloaded sdk.